charts/wasp-thing-service-v2/templates/deployment.yaml

```diff
@@ -0,0 +1,234 @@
+      {{- if .Values.priorityClassName }}
+      priorityClassName: {{ .Values.priorityClassName | quote }}
+      {{- end }}
+      {{- if .Values.schedulerName }}
+      schedulerName: {{ .Values.schedulerName | quote }}
+      {{- end }}
+      {{- if .Values.topologySpreadConstraints }}
+      topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.topologySpreadConstraints "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.terminationGracePeriodSeconds }}
+      terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
+      {{- end }}
+      {{- if or .Values.initContainers .Values.initDbMigrate.enable (and (not .Values.postgresql.enabled) .Values.externalDatabase.create) }}
+      initContainers:
+        {{- if and (not .Values.postgresql.enabled) .Values.externalDatabase.create }}
+        - name: database-create
+          image: {{ template "wasp-thing-service.initDbCreate.image" . }}
+          imagePullPolicy: {{ .Values.initDbCreate.image.pullPolicy }}
+          env:
+            - name: DB_HOST
+              value: {{ template "wasp-thing-service.databaseHost" . }}
+            - name: DB_PORT
+              value: {{ template "wasp-thing-service.databasePort" . }}
+            - name: DB_NAME
+              value: {{ template "wasp-thing-service.databaseName" . }}
+            - name: DB_USERNAME
+              value: {{ template "wasp-thing-service.databaseUser" . }}
+            - name: DB_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "wasp-thing-service.databaseSecretName" . }}
+                  key: {{ include "wasp-thing-service.databaseSecretPasswordKey" . }}
+            - name: DB_POSTGRES_USERNAME
+              value: {{ .Values.externalDatabase.postgresqlPostgresUser }}
+            - name: PGPASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "wasp-thing-service.databaseSecretName" . }}
+                  key: {{ include "wasp-thing-service.databaseSecretPostgresPasswordKey" . }}
+          command:
+            - '/bin/sh'
+            - '-c'
+            - |
+              CONN_FLAGS="-h $DB_HOST -p $DB_PORT -U $DB_POSTGRES_USERNAME";
+              if [ "$( psql $CONN_FLAGS -tAc "SELECT 1 FROM pg_database WHERE datname=\$\$$DB_NAME\$\$" )" = '1' ]
+              then
+                echo "Database already exists doing nothing";
+                exit 0;
+              fi;
+              createuser $CONN_FLAGS $DB_USERNAME;
+              PSQL_ESC=$(echo $DB_PASSWORD | sed 's/./_/g');
+              psql $CONN_FLAGS -c "ALTER ROLE $DB_USERNAME WITH PASSWORD \$$$PSQL_ESC\$$$DB_PASSWORD\$$$PSQL_ESC\$";
+              createdb $CONN_FLAGS -O $DB_USERNAME $DB_NAME;
+        {{- end }}
```

Comment:
This is a new job purely to create the DB on an external Postgres should the DB not have already been created?

Reply:
Correct. The idea here is you might want to deploy a single database for all of WASP. Here it will create the database and user if it doesn't exist